native/src/main/java/io/ballerina/lib/data/csvdata/csv/Native.java

```diff
@@ -76,12 +77,18 @@ public static Object parseStream(Environment env, BStream csv,
                                      BMap<BString, Object> options, BTypedesc type) {
         try {
             final BObject iteratorObj = csv.getIteratorObj();
-            final Future future = env.markAsync();
-            DataReaderTask task = new DataReaderTask(env, iteratorObj, future, type,
-                    CsvConfig.createParseOptions(options),
-                    options.getStringValue(Constants.ConfigConstants.ENCODING));
-            DataReaderThreadPool.EXECUTOR_SERVICE.submit(task);
-            return null;
+            return env.yieldAndRun(() -> {
+                CompletableFuture<Object> future = new CompletableFuture<>();
+                DataReaderTask task = new DataReaderTask(env, iteratorObj, future, type,
+                        CsvConfig.createParseOptions(options),
+                        options.getStringValue(Constants.ConfigConstants.ENCODING));
+                DataReaderThreadPool.EXECUTOR_SERVICE.submit(task);
```

Comment:
Shall we remove the pool and use java 21 virtual thread

Reply:
Fixed in 7b4de1f18d8fc8a7c4349dfe297d09b03fa42d45